pkg/controller/cloneset/update/cloneset_update.go

```diff
@@ -141,8 +141,9 @@ func calculateUpdateCount(coreControl clonesetcore.Control, strategy appsv1alpha
 	}
 	waitUpdateIndexes = waitUpdateIndexes[:(len(waitUpdateIndexes) - partition)]
 
+	roundUp := strategy.MaxSurge == nil
```

Comment:
My fault. It is meaningless to compare with `nil`, because we have set the `maxSurge` defaults to `0` 

https://github.com/openkruise/kruise/blob/master/pkg/apis/apps/v1alpha1/defaults.go#L223

Reply:
so we need to check whether maxSurge is 0, I will update it.